### ISSUE TITLE
Mi Security 2.4.7 [Android App] [Ads]

### ISSUE THREAD

**Issue by Nzzyme** [AdguardTeam/AdguardFilters#44491]
Ads seen in the 'Security' (built-in) app of my phone (Xiaomi Redmi Note 3) running Android version 6.0.1 MMB29M

AG Version Used: v.3.2.150

Filters: AG Base, Mobile Ads, Peter Lowe's, AG Tracking Protection, AG Social Media, Malware Domains Blocklist, Spam404, NoCoin Filter List, AG Annoyances filter, Adblock Warning Removal List

Settings: Block ads in all apps is enabled, Filtering method is High

Stealth Mode is enabled and Phishing and malware protection is enabled

HTTPS Filtering is enabled & Filtering Method is Local VPN

Proxy is disabled

DNS Filtering is enabled. AG DNS with AG Simplified domain names filter enabled

Security app version is 2.6.7 (https://www.apkmirror.com/apk/xiaomi-inc/miui-security/miui-security-2-4-7-release/)

**Comment by Nzzyme** [comment]
Additional Question: Why are ads not blocked in this app? I don't see any log too for this app. Does AG not filter this?

**Comment by krystian3w** [comment]
Maybe embeded in app a few ads or app encrypt ads?

**Comment by Alex-302** [comment]
Screenshot, please.

**Comment by Alex-302** [comment]
Cannot install on Sony.

**Comment by Nzzyme** [comment]
https://ibb.co/5sgdQPY

Here you go

**Comment by Alex-302** [comment]
@AdguardTeam/filters-maintainers guys, do you have Xiaomi device?

**Comment by Alex-302** [comment]
@Nzzyme we need package name.

**Comment by BlazDT** [comment]
Only have Pixel.

**Comment by Nzzyme** [comment]
> @Nzzyme we need a package name.

com.miui.securitycenter

**Comment by Alex-302** [comment]
Ads can be loaded using their system app.
Try to search for `graph.facebook.com` in filtering log and look on souce.

**Comment by Nzzyme** [comment]
> Ads can be loaded using their system app.
> Try to search for `graph.facebook.com` in filtering log and look on souce.

All occurrences of graph.facebook.com are already blocked at my end.

**Comment by Alex-302** [comment]
Check with disabled Internet.

**Comment by Alex-302** [comment]
> Traffic from that app (and possibly others) is not passing through AG.

as I said above, ads can be loaded by anotrer app, or built-in to the app.

**Comment by Nzzyme** [comment]
Without internet, I don't see the ads. By the way, these are sometimes video ads too so internet is definitely needed for them to load. 

Loaded by another app? How do we fix this?

**Comment by Alex-302** [comment]
I heed a HAR file.

**Comment by Nzzyme** [comment]
Sent HAR file to dev@adguard.com

**Comment by krystian3w** [comment]
In zip into this topic.

**Comment by Nzzyme** [comment]
[adguard.zip](https://github.com/AdguardTeam/AdguardFilters/files/3905175/adguard.zip)
Here you go

**Comment by Alex-302** [comment]
I see only two requests
```
connectivitycheck.gstatic.com
messenger-noneu.truecaller.com
```

**Comment by Nzzyme** [comment]
Yes, the steps I followed were:
1. Disconnect from internet and clear cache & data from Security app
2. Exit AG and restart AG and turn HAR on
3. Turn internet on and AG on
4. Run the Security app and perform the scan operation. Ads shown.
5. Turn HAR option off.
6. Export and zip HAR file

As I said earlier, the Ad traffic can't be seen in Filtering Log. I am able to see traffic from other apps though in the log.

**Comment by Alex-302** [comment]
I don't know how to block it without device. I will ask someone.

**Comment by krystian3w** [comment]
```
||connectivitycheck.gstatic.com^$app=com.miui.securitycenter
||messenger-noneu.truecaller.com^$app=com.miui.securitycenter
```

❓

**Comment by Nzzyme** [comment]
Could it be one to include in this list then? 

https://forum.adguard.com/index.php?threads/apps-where-adguard-for-android-cant-filter-and-or-remove-the-ads-from.25182/

**Comment by Nzzyme** [comment]
Just checking if there is any update on this?

**Comment by krystian3w** [comment]
Maybe nope.

---

Try hard reset phone, install AdGuard, activate protection with lists + **DNS** and now first open `Mi Security`?

**Comment by Alex-302** [comment]
Try to disable Internet for this app.
Also you can try to record HAR file
https://kb.adguard.com/en/android/solving-problems/har

**Comment by Nzzyme** [comment]
I tried to disable mobile data & WiFi for this app through AG (Apps Management) and that did not work too. I still saw ads. I had already sent you a HAR file which was of no use I suppose. For some reason, ad traffic from this app does not route through AG. 

There is a setting in the Security app called 'Receive Recommendations' which was enabled. I disabled that and no longer see ads. However, the question still remains as to why is AG unable to filter traffic from this app?

Who knows what else is communicating in the background? I thought that the entire device traffic is being routed through AG. Is this not the case?
